libpod/container_internal.go

```diff
@@ -1848,7 +1848,7 @@ func (c *Container) cleanupStorage() error {
 		if c.valid {
 			if err := c.save(); err != nil {
 				if cleanupErr != nil {
-					logrus.Errorf("Unmounting container %s: %v", c.ID(), cleanupErr)
+					logrus.Errorf("Unmounting container %s: %v", c.ID(), err)
```

Comment:
These changes do not look right to me, it will now report the same error twice and throw away the previous cleanupErr, it checks for `cleanupErr != nil` then you print `err` and set `cleanupErr = err` so cleanupErr is never used

Reply:
As the code is now, all errors are printed and the last one reported back.  I agree with you this is not ideal, I was not sure why it was done this way so I have not touched it but I can fix this pattern so the first error is reported to the caller while other ones are printed

Reply:
yes but I am not saying you have to change the pattern, just keep at as is right now which means print cleanupErr and not err.

Reply:
that seems wrong to me, we are printing a different error that is not related to the current failure.

I've pushed a new version where I change the pattern, since we both agree it is not ideal to print twice the same error.

Reply:
fine with we, definitely better then the current version.